### ISSUE TITLE
except list hides some methods of included types

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#7249]
### Summary of Problem
If two types have a method with the same name, and one of those types is excluded from a scope via the except list of a use statement, this accidentally prevents the included type from utilizing their method with the same name in the scope where the use is defined.  Ugh.

### Steps to Reproduce

**Source Code:**

```chapel
module M {
  class C1 {
    var x: int;

    proc foo() {
      writeln(x);
    }
  }

  class C2 {
    var y: bool;

    proc foo() {
      writeln(y);
    }
  }
}

module OtherMod {
  use M except C2;

  proc main() {
    var c1 = new C1(7);
    // The call below should work, but doesn't due to C2 having a method with the same name
    c1.foo();
    delete c1;
  }
}
```

**Compile command:**
`chpl foo.chpl`

**Associated Future Test(s):**
[`test/visibility/except/multipleTypesSameMethodName.chpl`](https://github.com/chapel-lang/chapel/blob/master/test/visibility/except/multipleTypesSameMethodName.chpl) (#7250)

Affected tests under ``--force-initializers``:
- [ ] visibility/only/typeIncludedAccessField
- [ ] visibility/only/typeIncludedAccessMethod
- [ ] visibility/rename/renameClass
- [ ] visibility/rename/renameRecord

### Configuration Information

- Output of `chpl --version`: 1.16.0 (pre-release)

**Comment by lydia-duncan** [comment]
I know exactly where the fix for this should go, I think.

**Comment by lydia-duncan** [comment]
I was originally going to punt on this, but since initializers all share the same name, I'm realizing this bug means that init and _new will not be found for other types if any type that uses initializers is excluded, so this will need to get fixed sooner rather than later

**Comment by benharsh** [comment]
I'll document a few of the related problems/concerns here:

1) We should have a way to ask a UseStmt if a method call is visible. This requires passing the type name and method name to a UseStmt method.

2) Putting all 'relatedNames' into a list loses information (where they came from). I think it would be better if there was a map from type-name --> set-of-member-names. I'm not sure if that's possible within the current UseStmt/scopeResolve architecture.

3) During type resolution (e.g. resolveNormalCall), we will want to ask whether a method call is visible. If I ask the UseStmt if "T.foo" is visible, but "T" is renamed, the UseStmt will say it is not visible. It appears as though we only store a map from new-name --> old-name, so it's not a simple lookup. Especially if the new names are still floating around.

**Comment by benharsh** [comment]
4. "only" may have bugs as well. For example, we should make sure we can handle compiler-recognized functions like ``_new`` and ``_defaultOf`` and not consider them 'invisible'.

**Comment by bradcray** [comment]
Scanning this issue quickly (I didn't recall it prior to this week's activity), it seems as though #7857 may also be related to it?  I could be wrong about that, however.

**Comment by bradcray** [comment]
Scanning gating issues today, I was surprised that this one was marked as such.  Discussing with Lydia, we decided that it's more "very concerning" than "gating", so I'm removing that label.

**Comment by lydia-duncan** [comment]
Closed by #10439
